CK3toEU4/Source/EU4World/Output/outNationalIdeas.cpp

```diff
@@ -0,0 +1,34 @@
+#include "outNationalIdeas.h"
+#include "OSCompatibilityLayer.h"
+
+std::ostream& EU4::operator<<(std::ostream& output, const NationalIdeas& idea)
+{
+	output << idea.getDynamicName() + "_ideas = {";
+
+	// National Traditions
+	output << "\n\tstart = {";
+	for (auto& effect: idea.getEthosEffects())
+		output << "\n\t\t " + effect.type + " = " + effect.value;
+	output << "\n\t}";
+
+	// Trigger
+	output << "\n\ttrigger = {\n\t\tculture = " + idea.getDynamicName() + "\n\t}\n";
+	output << "\tfree = yes\n\n";
+
+	// First 7 traditions become ideas, last one becomes ambition
+	for (int i = 0; i < idea.getTraditionIdeas().size() - 1; i++)
+	{
+		output << "\t" + idea.getTraditionIdeas()[i] + " = {";
+
+		for (auto& effect: idea.getTraditionEffects()[i])
+			output << "\n\t\t" + effect.type + " = " + effect.value;
+
+		output << "\n\t}\n";
+	}
+
+	// National Ambition
+	output << "\tbonus = {\n\t\t" + idea.getTraditionEffects().back().front().type + " = " + idea.getTraditionEffects().back().front().value + "\n\t}\n";
```

Comment:
For clarity it's probably better to use one or more intermediate variables. If I understand the .back() part that's grabbing the eighth tradition, but I'm not sure what the .front() accomplishes.

Reply:
The bonus for finishing National Ideas, the National Ambition is only supposed to have one effect. So the .front() is to get just one effect if the tradition was mapped to a multi-effect idea. A -10% advisor cost and +1 Possible Advisor idea would become just a -10% advisor cost Ambition. But yeah, I'll use some intermediates for some of those.